src/cdomains/congruenceClosure.ml

```diff
@@ -269,7 +269,7 @@ module Disequalities = struct
       let l2 = map_find_all t2 arg in
       fold_left2 add_diseq acc l1 l2
     in
-    List.fold f acc (TSet.to_list tset)
+    List.fold_left f acc (TSet.elements tset)
```

Comment:
Why is this (and maybe `check_neq` above) working on lists at all? It's creating a list from the set and then folding the list. Obviously this could be on `Seq`, but there's possibly an even more direct way: just `TSet.fold`.

Reply:
I changed it here. For `check_neq`, we're missing a `ZMap.fold2`, so it's not so clear how to do it. I'd suggest leaving it as-is.